profile/src/wallet/wallet_accounts.rs

```diff
@@ -0,0 +1,561 @@
+use crate::prelude::*;
+
+impl Wallet {
+    /// Adds a device factor source to Profile and SecureStorage, this method will only
+    /// return `Ok` if both the mnemonic was successfully saved to SecureStorage and the
+    /// DeviceFactorSource present in Profile and Profile also successfully updated in
+    /// SecureStorage.
+    ///
+    /// Returns `Err` if it is already present in Profile. It is Wallet Client
+    /// dependent if it throws if already present in SecureStorage.
+    ///
+    /// If saving of `MnemonicWithPassphrase` to SecureStorage succeeds, but adding
+    /// `DeviceFactorSource` to Profile/saving of Profile to SecureStorage fails, then
+    /// this method will try to remove the newly saved `MnemonicWithPassphrase` from
+    /// `SecureStorage`.
+    ///
+    /// Takes ownership of `PrivateHierarchicalDeterministicFactorSource`
+    pub fn add_private_device_factor_source(
```

Comment:
naming? `save` instead of `add`? hmm, tricky because this method saves the mnemonic into secure storage AND saves the DeviceFactorSource into Profile (which also takes a snapshot of Profile and saves it to secure storage).

Reply:
`add` feels ok to me.